source/Halibut.TestProxy/HttpProxyService.cs

```diff
@@ -53,6 +53,12 @@ public async Task StartAsync()
                 {
                     logger.LogError(ex, "An error has occurred running the HTTP proxy service");
                 }
+                finally
+                {
+                    // Ensure that StartAsync can complete,
+                    // regardless of whether there was an error or not
+                    started = true;
```

Comment:
I think `StartAsync` should throw if it fails to start the listener e.g. start is never set to true by `StartInternalAsync`

e.g.

```
public async Task StartAsync()
{
            Exception? startupException = null;
            var _ = Task.Run(async () =>
            {
                try
                {
                    await StartInternalAsync();
                }
                catch (Exception ex)
                {
                    startupException = ex;
                    logger.LogError(ex, "An error has occurred running the HTTP proxy service");
                }
            });

            while (!started)
            {
               cancellationTokenSource.Token.ThrowIfCancellationRequested();
            
                if(startupException != null)
                {
                    throw startupException;
                }
            
                await Task.Delay(TimeSpan.FromMilliseconds(10), cancellationTokenSource.Token);
            }
}
```

Reply:
That's nicer for sure, no worries.